app/extensions/brave/content/scripts/windowData.js

```diff
@@ -0,0 +1,14 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+(function () {
+  var ipcRenderer = chrome.ipc;
+
+  var data = {
+    "title": document.title,
+    "url": window.location.href
```

Comment:
do we need the url here? The title is already set to the url if `page-title-updated` isn't fired. I can't articulate a particular threat, but I just feel like it would be generally safer to use the url from frames.js rather than sending back `window.location.href` @diracdeltas ?